### ISSUE TITLE
build tests segfault on UV 300 running SLES 12 SP2

### ISSUE THREAD

**Issue by dominikborkowski** [xianyi/OpenBLAS#1545]
Trying to build OpenBLAS 0.2.20 on SGI UV 300 running SLES 12 SP2 and running into problems, hope somebody may have a clue as to what's going on. TL;DR: tests segfault when using more than one thread.

First run, without BIGNUMA=1 results in the following:
1) OPENBLAS_NUM_THREADS=1 OMP_NUM_THREADS=1 with sblat1, dblat1, cblat1, zblat1 tests succeed
2) OPENBLAS_NUM_THREADS=2 ./sblat1 fails with 
OpenBLAS Warning : The number of CPU/Cores(432) is beyond the limit(256). Terminated.

Following this error, next build is executed with 'BIGNUMA=1 make', this results in the following:

1) OPENBLAS_NUM_THREADS=1 OMP_NUM_THREADS=1 with sblat1, dblat1, cblat1, zblat1 tests succeed
2)OPENBLAS_NUM_THREADS=2 ./sblat1
Makefile:8: recipe for target 'level1' failed
make[1]: *** [level1] Segmentation fault (core dumped)
make[1]: *** Waiting for unfinished jobs....

At this point trying to execute any of the tests (sblat1/dblat1/cblat1/zblat1) with either no OPENBLAS/OMP env variables set, or by setting them to use more than one thread results in segmentation fault. Example strace is available at https://gist.github.com/dominikborkowski/c44b554ab94e5e6dd336d174ee25e1c0

Earlier versions of OpenBLAS compile just fine. Has anybody else run into this problem?

Thanks!

**Comment by martin-frbg** [comment]
Can you try with current develop snapshot please, there have been numerous fixes since 0.2.20 ? (And I assume you did do `make clean` between builds with and without BIGNUMA=1 ?)

**Comment by dominikborkowski** [comment]
Yes, I made sure to clean the environment in between the attempts, even by completely removing the source tree.

And looks like using latest tree from git builds without any issues, both with and without BIGNUMA=1

```
 OpenBLAS build complete. (BLAS CBLAS LAPACK LAPACKE)

  OS               ... Linux
  Architecture     ... x86_64
  BINARY           ... 64bit
  C compiler       ... GCC  (command line : cc)
  Fortran compiler ... GFORTRAN  (command line : gfortran)
  Library Name     ... libopenblas_haswellp-r0.3.0.dev.a (Multi threaded; Max num-threads is 432)
```
[...]
```
OPENBLAS_NUM_THREADS=432 OMP_NUM_THREADS=432 ./cblat1
 Complex BLAS Test Program Results


 Test of subprogram number  1            CDOTC
                                    ----- PASS -----

 Test of subprogram number  2            CDOTU
                                    ----- PASS -----

 Test of subprogram number  3            CAXPY
                                    ----- PASS -----

 Test of subprogram number  4            CCOPY
                                    ----- PASS -----

 Test of subprogram number  5            CSWAP
                                    ----- PASS -----

 Test of subprogram number  6            SCNRM2
                                    ----- PASS -----

 Test of subprogram number  7            SCASUM
                                    ----- PASS -----

 Test of subprogram number  8            CSCAL
                                    ----- PASS -----

 Test of subprogram number  9            CSSCAL
                                    ----- PASS -----

 Test of subprogram number 10            ICAMAX
                                    ----- PASS -----
```

Thank you! I'll use the dev snapshot then. Cheers!
